src/main/java/org/ohdsi/webapi/security/PermissionController.java

```diff
@@ -159,7 +161,16 @@ public void grantEntityPermissionsForRole(
             AccessRequestDTO accessRequestDTO
     ) throws Exception {
 
+        // check if user, or his "teamProject" own the entity being given access:
         permissionService.checkCommonEntityOwnership(entityType, entityId);
+        // furthermore, check if the entity is being shared with "public" role (roleId==1). If yes, then
+        // check if user has the necessary global/public sharing permission ("artifact:global:share:put") to do so:
+        if (roleId == RoleEntity.PUBLIC_ROLE_ID) {
```

Comment:
I am unsure on this check, in WebAPI everyone will get `public` permissions right now. Is this not the case for Team Projects?

Reply:
it is about the last case.

Reply:
I'll improve the comment.